### ISSUE TITLE
Allow for multi instance log in

### ISSUE THREAD

**Issue by rafapirotto** [Sonarr/Sonarr#7199]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

Basically, I have multiple instances of Sonarr and whenever I'm logged in to one and then try to log in to the other one it basically logs me out from the first one. So login in one instance logs me out from the other one. I think that the reason is that the cookie is named "SonarrAuth" in both instances so one replaces the other one, which would explain why I get redirected to the log in screen:
![sonarr-bug](https://github.com/user-attachments/assets/72a7fee6-1af5-43fe-9e10-4ea99ec44a25)

### Describe the solution you'd like

I think the solution is as simple as using "`SonarrInstanceNameAuth`" to prevent the instances from replacing the other's cookie hence login out from the other instance. This can be applied to the other settings in localStorage as well.
![image](https://github.com/user-attachments/assets/0b250f8f-2a46-4bc0-b68a-1a2f50f1ec9f)

This could be `SonarrInstanceNameAuth` instead of `SonarrAuth`:
![image](https://github.com/user-attachments/assets/cbdd60e6-6364-4069-b89d-03a4457538ed)

### Describe alternatives you've considered

Just live with this. Not much else to add to this section.

**Comment by rafapirotto** [comment]
Maybe the change is something along these lines:
`src/Sonarr.Http/Authentication/AuthenticationBuilderExtensions.cs`
```
        public static AuthenticationBuilder AddAppAuthentication(this IServiceCollection services)
        {
            return services.AddAuthentication()
                .AddNone(AuthenticationType.None.ToString())
                .AddExternal(AuthenticationType.External.ToString())
                .AddBasic(AuthenticationType.Basic.ToString())
                .AddCookie(AuthenticationType.Forms.ToString(), options =>
                {
                    // update this to use the instanceName variable
                    options.Cookie.Name = "SonarrAuth";
                    options.AccessDeniedPath = "/login?loginFailed=true";
                    options.LoginPath = "/login";
                    options.ExpireTimeSpan = TimeSpan.FromDays(7);
                    options.SlidingExpiration = true;
                    options.ReturnUrlParameter = "returnUrl";
                })
                .AddApiKey("API", options =>
                {
                    options.HeaderName = "X-Api-Key";
                    options.QueryName = "apikey";
                })
                .AddApiKey("SignalR", options =>
                {
                    options.HeaderName = "X-Api-Key";
                    options.QueryName = "access_token";
                });
        }
```

**Comment by augustuen** [comment]
IIRC, a fix for this (at least until a change is made) is to use different subdomains rather than base URLs for each instance. 

I see in your screenshots you're using a local IP, in which case you could also disable auth for local adresses.

**Comment by rafapirotto** [comment]
Don't I need to setup a reverse proxy to use different subdomains?  (Sorry for the ignorance). Also, don't you agree that the suggested change is fairly easy and straightforward? And that would (I believe) solve the issue completely. What are your thoughts?

**Comment by bakerboy448** [comment]
Ref - https://github.com/Radarr/Radarr/issues/10416
Ref - Qstick's prior investigation on this issue https://github.com/Radarr/Radarr/pull/6722

TL;DR if it was a simple, quick fix - it would be done already....it's not

**Comment by markus101** [comment]
After investigating further, it's possible, but unsurprisingly it's not really "easy and straightforward" as "update this to use the instanceName variable".

**Comment by rafapirotto** [comment]
Anything I can do to help?
